ui/redux/actions/stripe.js

```diff
@@ -119,72 +143,41 @@ export const doListAccountTransactions = () => async (dispatch: Dispatch) =>
-      environment: stripeEnvironment,
-    },
-    'post'
-  ).then((accountLinkResponse: StripeAccountLink) =>
-    dispatch({ type: ACTIONS.SET_ACCOUNT_LINK, data: accountLinkResponse })
-  );
+export const doGetCustomerStatus = () => async (dispatch: Dispatch, getState: GetState) => {
+  const state = getState();
+  const isFetching = selectCustomerStatusFetching(state);
+
+  if (isFetching) return;
 
-export const doGetCustomerStatus = () => async (dispatch: Dispatch) => {
   dispatch({ type: ACTIONS.SET_CUSTOMER_STATUS_STARTED });
 
   return await Lbryio.call('customer', 'status', { environment: stripeEnvironment }, 'post')
-    .then((customerStatusResponse) => {
+    .then((customerStatusResponse: StripeCustomerStatus) => {
       dispatch({ type: ACTIONS.SET_CUSTOMER_STATUS, data: customerStatusResponse });
 
       return customerStatusResponse;
     })
-    .catch((e) => {
-      dispatch({ type: ACTIONS.SET_CUSTOMER_STATUS, data: null });
-
-      return e;
-    });
-};
-
-export const doCustomerSetup = () => async (dispatch: Dispatch) =>
-  await dispatch(doGetCustomerStatus())
-    .then(() =>
-      Lbryio.call(
-        'customer',
-        'setup',
-        { environment: stripeEnvironment },
-        'post'
-      ).then((customerSetupResponse: StripeCustomerSetupResponse) =>
-        dispatch({ type: ACTIONS.SET_CUSTOMER_SETUP_RESPONSE, data: customerSetupResponse })
-      )
-    )
     .catch((error) => {
-      // errorString passed from the API (with a 403 error)
-      const errorString = 'user as customer is not setup yet';
-
-      // if it's beamer's error indicating the account is not linked yet
-      // $FlowFixMe
-      if (error.message && error.message.indexOf(errorString) > -1) {
-        // get a payment method secret for frontend
-        Lbryio.call(
-          'customer',
-          'setup',
-          { environment: stripeEnvironment },
-          'post'
-        ).then((customerSetupResponse: StripeCustomerSetupResponse) =>
-          dispatch({ type: ACTIONS.SET_CUSTOMER_SETUP_RESPONSE, data: customerSetupResponse })
-        );
-        // 500 error from the backend being down
-      } else if (error === 'internal_apis_down') {
+      if (error === 'internal_apis_down') {
         doToast({ message: __(STRIPE.APIS_DOWN_ERROR_RESPONSE), isError: true });
       } else {
         // probably an error from stripe
         doToast({ message: __(STRIPE.CARD_SETUP_ERROR_RESPONSE), isError: true });
       }
+
+      dispatch({ type: ACTIONS.SET_CUSTOMER_STATUS, data: null });
+
+      return error;
     });
+};
+
+export const doCustomerSetup = () => async (dispatch: Dispatch) =>
+  await Lbryio.call('customer', 'setup', { environment: stripeEnvironment }, 'post').then(
+    (customerSetupResponse: StripeCustomerSetupResponse) => {
+      dispatch({ type: ACTIONS.SET_CUSTOMER_SETUP_RESPONSE, data: customerSetupResponse });
+      return customerSetupResponse;
+    }
+  );
```

Comment:
:small_red_triangle: No error handling.  Spins forever.  There are few other news ones like this.

<img width="309" alt="image" src="https://user-images.githubusercontent.com/64950861/197174498-ca4776cb-2e62-4c01-8444-6ba1c9101788.png">